### PR TITLE
Add network storage specific doc

### PR DESCRIPTION
**Pull request by nick-stroud**
### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: docs/network_storage.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+The HPC Toolkit provides powerful tools for working with network
+storage.
+
+The Toolkit contains modules that will **provision**:
+
+- [Filestore (GCP managed nfs)][filestore]
+- [DDN EXAScaler lustre][ddn-exascaler]
+- [Intel DAOS][intel-daos]
+- [Nfs server (non-GCP managed)][nfs-server]
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/network_storage.md]: Done

File: docs/network_storage.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# Network Storage in the HPC Toolkit
+
+The HPC Toolkit provides powerful tools for working with network
+storage.
+
+The Toolkit contains modules that will **provision**:
+
+- [Filestore (GCP managed nfs)][filestore]
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/network_storage.md]: Done.

File: docs/network_storage.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+- nfs
+- lustre
+- gcsfuse
+
+## Connecting to Network Storage
+
+In addition to provisioning a network storage device, most file system modules
+contain scripts that will install any required software needed to utilize the
+device and mount the file system automatically.
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/network_storage.md]: Updated. Slightly different than proposed verbiage.

File: docs/network_storage.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    use: [network1]
+    settings: {local_mount: /home}
+
+  - id: workstation
+    source: ./modules/compute/vm-instance
+    use: [network1, homefs]  # Note this line
+```
+
+The example above is creating a filestore and automatically mounting it to a VM.
+Take note of the line with the comment saying `# Note this line`. By adding
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/network_storage.md]: Done.

File: docs/network_storage.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+- The `homefs` filestore outputs scripts for nfs client installation and
+  mounting.
+- The `workstation` VM reads these scripts and generates a startup script for
+  the VM.
+- The VM startup script will automatically install the nfs client on its first
+  boot (if not already installed).
+- The VM will add `homefs` to fstab and mount the file system.
+
+This same pattern works across most modules in the toolkit. The compatibility
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/network_storage.md]: Done.

File: docs/network_storage.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+> **_NOTE:_** The exact names of the runners may be different from module to
+> module.
+
+## Compatibility Matrix
+
+The following matrix shows the best method by which each type of network storage
+device should be mounted to each mount capable module.
+
+. | Slurm V4 | Slurm V5 | Batch | vm-instance | Packer (client install) | HTCondor\*\* | PBS Pro**
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/network_storage.md]: Done.

**Review by heyealex — DISMISSED** [review]

**Review by cboneti — APPROVED** [review]

**Pull request merged**